eclipse-collections/src/main/java/org/eclipse/collections/impl/utility/Iterate.java

```diff
@@ -130,6 +131,10 @@ private Iterate()
         {
             ((InternalIterable<T>) iterable).forEach(procedure);
         }
+        else if (iterable instanceof BatchIterable)
+        {
+            ((BatchIterable<T>) iterable).forEach(procedure);
+        }
```

Comment:
We hit this path when running `map.keysView().toString()` since:

- keysView() is implemented by returning `LazyIterate.adapt(this.keySet())`
- the `LazyIterableAdapter` eventually delegates to `Iterate.forEach()`
- the keys/values/entries inner classes in UnifiedMap are BatchIterables, not RichIterables